ironic/drivers/modules/drac/raid.py

```diff
@@ -782,15 +867,18 @@ def delete_configuration(self, task):
         node = task.node
 
         controllers = list()
-        for disk in list_virtual_disks(node):
-            controller = dict()
-            controller_cap = delete_virtual_disk(node, disk.id)
-            controller['raid_controller'] = disk.controller
-            controller['is_reboot_required'] = controller_cap[
-                'is_reboot_required']
-            controllers.append(controller)
-
-        return _commit_to_controllers(node, controllers)
+        drac_raid_controllers = list_raid_controllers(node)
+        for cntrl in drac_raid_controllers:
+            if is_raid_controller(node, cntrl.id):
```

Comment:
You'll need to update the above call to pass the list of RAID controllers in so that we don't beat up the iDRAC with calls to enumerate DCIM_ControllerView.

Reply:
on line 870: the function is just a anchor or a bridge or simple wrapper which call the list_raid_controllers() from dracclient.

Reply:
Hey Rachit, I recently changed is_raid_controller() so that it will return True for both BOSS cards and PERC RAID Controllers (https://review.opendev.org/#/c/653850/).  As part of this change, I added an optional parameter to is_raid_controller() that allows you to pass in the list of controllers.  I did this because the code that checks to see if a controller is a BOSS controller has to issue a query to the iDRAC to get the list of controllers.  Since we frequently call is_raid_controller() in a loop as above, I didn't want us to be needlessly querying the iDRAC with repeated calls to get the list of controllers.  The end result is that you can pass in the list of controllers to is_raid_controller() and that will optimize things so that it will use the passed in list instead of querying the iDRAC.  To take advantage of this optimization, you will need to change line 872 above so that is passes the list of controllers to is_raid_controller(), and change that local method so that it passes the list along to client.is_raid_controller().  See https://github.com/openstack/python-dracclient/blob/master/dracclient/client.py#L1008

Reply:
Oh.! okay i got your point...Thanks